src/components/Filters/Filters.jsx

```diff
@@ -0,0 +1,218 @@
+          toggle && styles["display"]
+        }`}
+      >
+        <div className={`${styles["filters"]} ${styles["filter-header"]}`}>
+          <p>Filters</p>
+          <button
+            className={styles["clear-filter-link"]}
+            onClick={() => filterDispatch({ type: "CLEAR" })}
+          >
+            Clear
+          </button>
+          {toggle && (
+            <FontAwesomeIcon icon={faXmark} size="2x" onClick={toggleOption} />
+          )}
+        </div>
+        <div className={styles["filters"]}>
+          <p>Price</p>
+          <div className={styles["slider"]}>
+            <input
+              type="range"
+              min="0"
+              max="2000"
+              value={filterState.price}
+              step="100"
+              onChange={(e) =>
+                filterDispatch({ type: "PRICE", payload: e.target.value })
+              }
+            />
+            <p id="rangeValue">{filterState.price}</p>
+          </div>
+        </div>
+        <div className={styles["filters"]}>
+          <p>Filters</p>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={filterState.showOutOfStock}
+                onChange={() => filterDispatch({ type: "TOGGLE_STOCK" })}
+              />
+              Show Out of Stock
+            </label>
+          </div>
+        </div>
+        <div className={styles["filters"]}>
+          <p>Category</p>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={
+                  !filterState.category
+                }
+                onChange={() =>
+                  filterDispatch({
+                    type: "CATEGORY",
+                    payload: false,
+                  })
+                }
+              />
+              All
+            </label>
+          </div>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={
+                  filterState.category &&
+                  filterState.category === "Instant Coffee"
+                }
+                onChange={() =>
+                  filterDispatch({
+                    type: "CATEGORY",
+                    payload: "Instant Coffee",
+                  })
+                }
+              />
+              Instant Coffee
+            </label>
+          </div>
+          <div>
+            <label>
+              <input
+                type="checkbox"
+                checked={
+                  filterState.category &&
+                  filterState.category === "Ground Coffee"
+                }
+                onChange={() =>
+                  filterDispatch({
+                    type: "CATEGORY",
+                    payload: "Ground Coffee",
+                  })
+                }
+              />
+              Ground Coffee
+            </label>
+          </div>
+        </div>
```

Comment:
Please check this code. Unable to check multiple categories at once. Was this intentional because there are only 2 categories and 'All' in case we need both?

Reply:
Yes, it was intentionally done because there are only 2 categories as of now.